pandas/tests/indexing/test_interval.py

```diff
@@ -122,6 +148,46 @@ def test_with_slices(self):
         with pytest.raises(ValueError):
             s[0:4:2]
 
+    @pytest.mark.xfail(reason="new indexing tests for issue 16316")
+    def test_with_slices_updated_behavior(self):
+
+        s = self.s
+
+        # slice of interval
+        expected = s.iloc[4:]
+        result = s.loc[Interval(3, 4):]
+        tm.assert_series_equal(expected, result)
+
+        expected = s.iloc[4:]
+        result = s[Interval(3, 4):]
+        tm.assert_series_equal(expected, result)
+
+        with pytest.raises(KeyError):
+            s.loc[Interval(3, 6):]
+
+        with pytest.raises(KeyError):
+            s[Interval(3, 6):]
+
+        with pytest.raises(KeyError):
+            s.loc[Interval(3, 4, closed='left'):]
+
+        with pytest.raises(KeyError):
+            s[Interval(3, 4, closed='left'):]
+
+        with pytest.raises(KeyError):
+            s.loc[Interval(3, 4, closed='both'):]
+
+        with pytest.raises(KeyError):
+            s[Interval(3, 4, closed='both'):]
+
+        # slice of scalar
+        with pytest.raises(NotImplementedError):
+            s[0:4] ## not sure what the behvaior should be here.
+
+        # slice of scalar with step != 1
+        with pytest.raises(ValueError):
+            s[0:4:2] ## This should probably definitely fail I guess?
```

Comment:
see above, if we decide the positional slicing is fine, this should work as well

Reply:
@alexlenail im trying to figure out why this restriction exists.  do you remember why slices with step!=1 are disallowed?  cc @jorisvandenbossche

Reply:
@jbrockmendel can you give some context? Eg example code snippet (it's hard to see in the diff what the above is about)

Reply:
```
ser = pd.Series(np.arange(5), pd.IntervalIndex.from_breaks(np.arange(6)))

expected = ser.iloc[0:4:2]

result = ser[0:4][::2]  # <-- works
tm.assert_series_equal(result, expected)

result2 = ser[0:4:2]   # <-- raises ValueError ATM bc step!=1
```

It isn't clear to me why this last line should not behave like the other two slicings.

Reply:
Yeah, I don't think there is a good reason for that. 
I suppose maybe initially in this PR, getitem with integers was meant to do label-based slicing finding the containing intervals, and in such a case, it makes sense to disallow a step. But for positional indexing, this should just be allowed of course.

So the check should only be done depending on the `kind` I suppose.